docs/proposals/20200506-conditions.md

```diff
@@ -0,0 +1,363 @@
+
+- [Title](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [User Stories [optional]](#user-stories-optional)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+    - [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Design Details](#design-details)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+      - [Examples](#examples)
+        - [Alpha -> Beta Graduation](#alpha---beta-graduation)
+        - [Beta -> GA Graduation](#beta---ga-graduation)
+        - [Removing a deprecated flag](#removing-a-deprecated-flag)
+    - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+    - [Version Skew Strategy](#version-skew-strategy)
+  - [Implementation History](#implementation-history)
+  - [Drawbacks [optional]](#drawbacks-optional)
+  - [Alternatives [optional]](#alternatives-optional)
+  - [Infrastructure Needed [optional]](#infrastructure-needed-optional)
+
+## Glossary
+
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+Condition: The state of an object with regard to its appearance, quality, or working order.
+
+## Summary
+
+In Cluster API a workload cluster is composed of a fair number of Kubernetes objects, and to
+understand the status of your Cluster you have to jump from one object to another. Or, in the
+worst case, you have to look at the logs from the Cluster API pods to get more details on why
+things aren’t working.
+
+This proposal introduces the concept of “conditions” as a tool designed to provide “at glance”
+view of the status of a cluster, providing immediate answers and fine-grained guidance when
+investigating issues. 
+
+## Motivation
+
+- Cluster API requires a large number of objects to define a workload cluster: Cluster,
+  InfrastructureCluster, KubeadmControlPlane, MachineDeployments, Machines, and so on.
+  Each object carries its own status. Operators need to describe each object individually to
+  determine an overall cluster health summary.
+- Objects can be tied to each other directly (via object references), or indirectly
+  (via owner references). These dependencies can often require a deep knowledge of the
+  project’s internals and its APIs to investigate issues or problems. 
+- Some objects, like MachineSets or MachineDeployments, are considered a collection of
+  other objects (e.g. MachineSets). Their status is effectively tied to the status of each
+  item in the collection. 
+
+### Goals
+
+In order to make this proposal actionable in the v0.3.x timeframe, the proposal is going to
+be developed in an incremental fashion in three iterations. 
+
+The goal of this first iteration is:
+
+1. To agree on a design for the condition types.
```

Comment:
I think that to agree on a design for the condition types we need to include as goals a list of user stories traversing capi components which are not well communicated today but they will with the new conditions. Then in the "Implementation" section we can elaborate how we'll achieve these goals fleshing out where we set those conditions, which values and how the user can see them in a meaningful manner.

Reply:
I think we should preserve the distinction between goals and user stories, but I agree on expanding the implementation section with user stories and examples traversing CAPI components (see new commits)